core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -221,6 +223,7 @@ public void delete(PageId pageId) throws IOException, PageNotFoundException {
       try (LockResource r1 = new LockResource(mMetaLock.writeLock())) {
         mMetaStore.removePage(pageId);
       }
+      mEvictor.updateOnDelete(pageId);
```

Comment:
Should we abstract this logic to another method?

1. Grab metalock
2. Do metastore operation
3. Release metalock
4. Notify listeners

Reply:
what's the benefit in this case?

Reply:
Just makes sure that we do all the necessary steps, otherwise we might have cases where we forget to update listeners or grab locks